packages/datetime2/src/components/date-range-input3/dateRangeInput3.tsx

```diff
@@ -288,6 +290,18 @@ export class DateRangeInput3 extends DateFnsLocalizedComponent<DateRangeInput3Pr
     // Callbacks - DateRangePicker3
     // ===========================
 
+    private getTimePrecision = () => {
+        // timePrecision may be set as a root prop or as a property inside timePickerProps, so we need to check both
+        const { timePickerProps, timePrecision } = this.props;
+        const fromProps = timePickerProps?.precision ?? timePrecision;
+        if (fromProps != null) {
+            return fromProps;
+        }
+
+        // if timePickerProps is defined, the `TimePicker` will render with it's default `precision`
+        return timePickerProps != null ? TimePicker.defaultProps.precision : undefined;
```

Comment:
A bit confused by the comment above. Will the `TimePicker` still render with its default precision if `timePickerProps` is `undefined`?

I'm curious how important it is to return `TimePicker.defaultProps.precision` vs `undefined` here.

Reply:
because of this bit of existing behavior, which I tried to explain more here: https://github.com/palantir/blueprint/pull/6881/files#diff-d2a31417b184106fdf4054399ab49455504ff747c19714aa6df0bb7f8e26bf90R238-R239

let me know any suggestions you can think of to help clarify this

Reply:
if `timePickerProps` is not defined, the underlying `DateRangePicker` will have default `timePickerProps`, and therefor not show the time pickers - so in that case we say the precision is `undefined.

if `timePickerProps` is defined, we return the default `precision`, since at this point in this method we already would have returned an explicitly defined precision either directly on this component, or from the `timePickerProps`